src/app/dao/tests/FirebaseFirestoreDAOMock.spec.ts

```diff
@@ -25,6 +25,7 @@ export abstract class FirebaseFirestoreDAOMock<T extends FirebaseJSONObject> imp
     }
 
     public callbacks: [FirebaseCondition[], FirebaseCollectionObserver<T>][] = [];
+    private readonly subDAOs: MGPMap<string, IFirebaseFirestoreDAO<FirebaseJSONObject>> = new MGPMap();
```

Comment:
Vu que toutes les classes ont pas de subDAO, on mettrait pas ça ailleurs ?

Reply:
Ici ça mérite discussion.
  1. Actuellement, y'a que ChatDAO qui a des subDAO, mais à terme je pense que pour des raisons de sécus, pas mal d'autres DAOs devront avoir des subDAOs (très certainement les parties en tout cas).
  2. Où mieux mettre ça ? Dans le cas où y'a pas de subDAO, c'est un champ inutilisé donc pas très problématique. Ou alors in FirebaseFirestoreDAOWithSubDAOs? Et du coup quel est le type du subDAO, un FirebaseFirestoreDAO ou un FirebaseFirestoreDAOWithSubDAOs? Me semble que ça repousse le problème plus loin sans le résoudre
  3. Un élément sub-optimal qui est présent c'est que si t'accède au subDAO de disons 'chats/123/', la classe est cachée dans cette variable subDAOs pour éviter de la recréer à chaque accès. Sauf que quand t'en a plus besoin, elle sera pas supprimée. Après moulte partie, t'auras des subDAO pour 'chats/123', 'chats/456', 'chats/789' qui seront stockés. Ils font rien mais prennent de la mémoire. C'est peut-être négligeable (un DAO en mémoire c'est... une string il me semble), donc peut-être pas un problème.
  4.

Reply:
Si plus de jeux s'apprêtent à avoir ça c'est bon alors. Sinon outre que c'est un string, c'est en mock, et en test on doit deleter tout à la fin des tests ! Donc pourquoi ça resterais si on fait ça ?

Reply:
hm c'pas une question de jeu ici ? Sinon : c'est aussi dans le non mock, cf. ligne 47 de FirebaseFirestoreDAO.ts

Reply:
Je voulais dire plus de table! Tu peux résoudre.